mlflow/tracking/_tracking_service/client.py

```diff
@@ -340,21 +357,35 @@ def search_traces(
         page_token: Optional[str] = None,
         run_id: Optional[str] = None,
         model_id: Optional[str] = None,
+        sql_warehouse_id: Optional[str] = None,
     ) -> PagedList[Trace]:
+        is_databricks = is_databricks_uri(self.tracking_uri)
+
         def download_trace_extra_fields(trace_info: TraceInfo) -> Optional[Trace]:
             """
             Download trace data and assessments for the given trace_info and returns a Trace object.
             If the download fails (e.g., the trace data is missing or corrupted), returns None.
             """
             # Only the Databricks backend supports additional assessments; avoid making
             # an unnecessary duplicate call to GET trace_info if not necessary.
-            if is_databricks_uri(self.tracking_uri):
+            if is_databricks:
                 trace_info_with_assessments = self.get_trace_info(
                     trace_info.request_id, should_query_v3=True
                 )
                 trace_info.assessments = trace_info_with_assessments.assessments
+
             try:
-                trace_data = self._download_trace_data(trace_info)
+                if is_databricks and is_uuid(trace_info.request_id):
+                    trace_data = self.get_online_trace_details(
+                        trace_info.request_id,
+                        sql_warehouse_id=sql_warehouse_id,
+                        # TODO
+                        # source_inference_table=trace_info.request_id,
+                        # source_databricks_request_id=trace_info.request_id,
```

Comment:
@prithvikannan can we get `source_inference_table` and `source_databricks_request_id` from `trace_info`?